core/storage/btree.rs

```diff
@@ -734,41 +763,51 @@ impl BTreeCursor {
         page.write_u16(pointer_area_pc_by_idx - page.offset, pc);
 
         // update first byte of content area
-        page.write_u16(BTREE_HEADER_OFFSET_CELL_CONTENT, pc);
+        page.write_u16(PAGE_HEADER_OFFSET_CELL_CONTENT_AREA, pc);
 
         // update cell count
         let new_n_cells = (page.cell_count() + 1) as u16;
-        page.write_u16(BTREE_HEADER_OFFSET_CELL_COUNT, new_n_cells);
+        page.write_u16(PAGE_HEADER_OFFSET_CELL_COUNT, new_n_cells);
     }
 
+    /// Free the range of bytes that a cell occupies.
+    /// This function also updates the freeblock list in the page.
+    /// Freeblocks are used to keep track of free space in the page,
+    /// and are organized as a linked list.
     fn free_cell_range(&self, page: &mut PageContent, offset: u16, len: u16) {
+        // if the freeblock list is empty, we set this block as the first freeblock in the page header.
         if page.first_freeblock() == 0 {
-            // insert into empty list
-            page.write_u16(offset as usize, 0);
-            page.write_u16(offset as usize + 2, len);
-            page.write_u16(BTREE_HEADER_OFFSET_FREEBLOCK, offset);
+            page.write_u16(offset as usize, 0); // next freeblock = null
+            page.write_u16(offset as usize + 2, len); // size of this freeblock
+            page.write_u16(PAGE_HEADER_OFFSET_FREEBLOCK, offset); // first freeblock in page = this block
             return;
         }
         let first_block = page.first_freeblock();
 
+        // if the freeblock list is not empty, and the offset is less than the first freeblock,
+        // we insert this block at the head of the list
         if offset < first_block {
-            // insert into head of list
-            page.write_u16(offset as usize, first_block);
-            page.write_u16(offset as usize + 2, len);
-            page.write_u16(BTREE_HEADER_OFFSET_FREEBLOCK, offset);
+            page.write_u16(offset as usize, first_block); // next freeblock = previous first freeblock
+            page.write_u16(offset as usize + 2, len); // size of this freeblock
+            page.write_u16(PAGE_HEADER_OFFSET_FREEBLOCK, offset); // first freeblock in page = this block
             return;
         }
 
+        // if we clear space that is at the start of the cell content area,
+        // we need to update the cell content area pointer forward to account for the removed space
+        // FIXME: is offset ever < cell_content_area?
         if offset <= page.cell_content_area() {
-            // extend boundary of content area
-            page.write_u16(BTREE_HEADER_OFFSET_FREEBLOCK, page.first_freeblock());
-            page.write_u16(BTREE_HEADER_OFFSET_CELL_CONTENT, offset + len);
+            page.write_u16(PAGE_HEADER_OFFSET_FREEBLOCK, page.first_freeblock()); // why is this here?
```

Comment:
?

Reply:
Looks like a noop

Reply:
Added a better FIXME for this. I don't want to change runtime code in this PR